src/vm.c

```diff
@@ -892,6 +906,7 @@ V7_PRIVATE val_t s_concat(struct v7 *v7, val_t a, val_t b) {
   }
   memcpy(s, a_ptr, a_len);
   memcpy(s + a_len, b_ptr, b_len);
+  s[a_len + b_len] = '\0';
```

Comment:
I'd keep it like that: that way we stress that regardless of the string type, it gets 0 terminated. Also, conditional would generate (arguably) longer and slower code. I've put a comment though.